packages/gatsby-plugin-benchmarks/gatsby-node.js

```diff
@@ -0,0 +1,124 @@
+const nodeFetch = require(`node-fetch`)
+
+const { performance } = require(`perf_hooks`)
+
+const bootstrapTime = performance.now()
+
+let apiServer
+let apiPath
+
+class BenchMeta {
+  constructor() {
+    this.flushing = false // Started flushing?
+    this.flushed = false // Completed flushing?
+    this.crashed = false
+    this.localTime = new Date().toJSON()
+    this.events = {
+      // TODO: we should also have access to node's timing data and see how long it took before bootstrapping this script
+      bootstrapTime,
+      instanceTime: performance.now(),
+      start: 0,
+      bootstrap: 0,
+      stop: 0,
+    }
+    this.started = false
+  }
+
+  markStart() {
+    if (this.started) {
+      this.crashed = "Error: Should not call markStart() more than once"
+      console.error("gatsby-plugin-benchmarks: " + this.crashed)
+      TODO // report error.
+    }
+    this.events.start = performance.now()
+    this.started = true
+  }
+
+  markDataPoint(name) {
+    this.events[name] = performance.now()
+  }
+
+  async markStop() {
+    if (!this.events.start) {
+      this.crashed =
+        "Error: Should not call markStop() before calling markStart()"
+      console.error("gatsby-plugin-benchmarks: " + this.crashed)
+      TODO // report error.
+    }
+    this.events.stop = performance.now()
+    return this.flush()
+  }
+
+  async flush() {
+    this.flushing = true
+
+    return nodeFetch(`https://${apiServer}${apiPath}`, {
+      method: `POST`,
+      headers: {
+        "content-type": `application/json`,
+        // "user-agent": this.getUserAgent(),
+      },
+      body: JSON.stringify({
+        time: this.localTime,
+        // This is a stub. We'll expand this as we determine what to log when running a benchmark
+        sessionId: JSON.stringify(this.events),
+      }),
+    })
+      .then(res => {
+        this.flushed = true
+        // Note: res.text returns a promise
+        return res.text()
+      })
+      .then(text => console.log("Server response:", text))
```

Comment:
I feel the actual return type of this function isn't what intended.
It seems that we actually return what is returned by the last then.

It might be clearer to await the two other promises 
```javascript
async flush() {
  this.flushing = true

  const resp = await nodeFetch(`https://${apiServer}${apiPath}`, {
    method: `POST`,
    headers: {
      "content-type": `application/json`,
      // "user-agent": this.getUserAgent(),
    },
    body: JSON.stringify({
      time: this.localTime,
      // This is a stub. We'll expand this as we determine what to log when running a benchmark
      sessionId: JSON.stringify(this.events),
    }),
  })
  if (resp.ok) {
    this.flushed = true
  }

  const text = await resp.text()
  console.log(`Server response:`, text)
}
```

Reply:
Haha, I looked at this for a long time and considered what you said. Then I shrugged and went for `undefined` instead. I weighed the added complexity of this function to the usefulness of the return value. Especially considering that the promise is more about a functional delay than the actual return value.

Although, to be fair, it might be used in the future to validate that the report lands, and to raise an alarm otherwise. So perhaps we should do this anyways.

Ok I'll refactor it such that the first `.then()` gets returned (the `.text()` promise).